### PR TITLE
2.6 Backport: Promtail: Add text encoding conversion to file targets (#6395)

### PR DESCRIPTION
**Pull request by DylanGuedes**
Adds support for text encoding conversion for file targets. To use it, add `encoding: <encoding_name>` into the scrapeconfig.

(cherry picked from commit f17d3d768cb915b656af20a2ef7973efae335d51)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -41,6 +41,7 @@ Here is the list with the changes that were produced since the previous release.
 * [5790](https://github.com/grafana/loki/pull/5790) **chaudum**: Add UDP support for Promtail's syslog target.
 * [6102](https://github.com/grafana/loki/pull/6102) **timchenko-a**: Add multi-tenancy support to lambda-promtail.
 * [6099](https://github.com/grafana/loki/pull/6099) **cstyan**: Drop lines with malformed JSON in Promtail JSON pipeline stage.
+* [6395](https://github.com/grafana/loki/pull/6395) **DylanGuedes**: Add encoding support
```
Comment (vlad-diachenko) [inline CHANGELOG.md:44]: could you please describe it more. and please add a dot at the end of the line

**Review by vlad-diachenko — CHANGES_REQUESTED** [review]
could you please fix the changelog

File: CHANGELOG.md (hunk 1 above)
Comment (slim-bean) [inline CHANGELOG.md:44]:
```suggestion
* [6395](https://github.com/grafana/loki/pull/6395) **DylanGuedes**: Add ability to configure custom encoding for non utf-8 log files.
```

**Review by vlad-diachenko — APPROVED** [review]
LGTM

File: CHANGELOG.md (hunk 1 above)
Comment (vlad-diachenko) [reply CHANGELOG.md]: @DylanGuedes could you please backport this commit to `main`

**Pull request merged**